prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -104,3 +131,22 @@ class DeployableMicrochainModifiableSystemPromptAgent3(
 ):
     task_description = AgentIdentifier.MICROCHAIN_AGENT_OMEN_LEARNING_3
     model = SupportedModel.llama_31_instruct
+
+
+class DeployableMicrochainWithGoalManagerAgent0(DeployableMicrochainAgent):
+    task_description = AgentIdentifier.MICROCHAIN_AGENT_OMEN_WITH_GOAL_MANAGER
+    model = SupportedModel.gpt_4o
+    goal_manager = GoalManager(
+        agent_id=task_description,
+        high_level_description="You are a trader agent in prediction markets to maximise your profit.",
+        agent_capabilities=(
+            "You are able to:"
+            "\n- List all binary markets that can be traded."
+            "\n- List the current outcome probabilities for each open market."
+            "\n- Predict the outcome probability for a market."
+            "\n- Buy, sell and hold outcome tokens in a market."
+            "\n- Query your wallet balance, and the positions you hold in open markets."
+            "\n- Query the past bets you've made, and their outcomes."
+        ),
```

Comment:
I'm not super-keen on how the `high_level_description` and `agent_capabilities` are defined here. It feels like some duplication of the system prompt, and `engine.help` respectively. But I couldn't think of a better way of doing this
- `high_level_description` is different enough from the system prompt, and I didn't think it was a good solution to try and derive it from the system prompt via an llm call
- `agent_capabilities` could be derived from `agent.engine.functions`, but there would be some messiness if I tried to define this after `build_agent_functions` is called.

Reply:
This is good enough, but maybe you can use (as you mentioned) `agent.engine.functions` (or `agent.help`) to get a long list of functions, and have an LLM generate a summary of those and output a bullet-point list?
At least you stay up-to-date with the most recent functions.

Reply:
Done 😃